### PR TITLE
offline search index functionality

### PR DESCRIPTION
**Pull request by opheliagame**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Build Eleventy
```
Comment (anantshri) [inline .github/workflows/build.yaml:1]: This is awesome. @oleeskild this is a working github pages build script here that you can use and the project can be published not just on vercel but github pages also and people can have choices. Not sure if you would like to keep this in the main repo but definately something that should go in the docs.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
```
Comment (anantshri) [inline .github/workflows/build.yaml:10]: my suggestion would be to look at 22.04 if possible.

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  
+    strategy:
+      matrix:
+        node-version: [16.x]
```
Comment (anantshri) [inline .github/workflows/build.yaml:14]: once you upgrade to 22.04 would suggest moving up to 18.x or 20.x https://endoflife.date/nodejs

**Review by anantshri — COMMENTED** [review]
I am not an official reviewer. spotted a few things so just lending a helping hand.

**Comment by denverbaumgartner** [comment]
hi @oleeskild , thank you for you excellent work so far. just checking through, it seems like github pages is still not supported? am I correct in understanding that the only blocker here is the 404 redirect? thanks!

**Comment by oleeskild** [comment]
Hi @denverbaumgartner 
Using GitHub Pages should be no problem. 

Here is an example build.yml that I've successfully used for publishing to Cloudflare. You should be able to switch the cloudflare task for a GitHub Pages task instead. 

```
name: Build and publish to Cloudflare

on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]

jobs:
  build:

    runs-on: ubuntu-22.04

    strategy:
      matrix:
        node-version: [20.x]
        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/

    steps:
    - uses: actions/checkout@v3
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v3
      with:
        node-version: ${{ matrix.node-version }}
        cache: 'npm'
    - run: npm install
    - run: npm run build --if-present

    - name: Publish to Cloudflare Pages
      uses: cloudflare/pages-action@v1
      with:
        apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
        accountId: {{myaccountid}}
        projectName: dgtestingflow
        directory: dist
        # Optional: Enable this if you want to have GitHub Deployments triggered
        gitHubToken: ${{ secrets.GH_TOKEN }}

        branch: main
    
        wranglerVersion: '3'
```

**Comment by denverbaumgartner** [comment]
Hey @oleeskild, thanks so much for the response. This has been helpful, but I am still running into some issues with deploying all aspects of the site directly through a `ci/cd` process. Is there an example of utilizing the plugin command `Digital Garden: Publish Multiple Notes` directly within an action?

**Pull request closed without merging**